BOJ/2231/dbql99.py

```diff
@@ -0,0 +1,10 @@
+target = int(input())
+
+for i in range(target):
+    num = sum(map(int, str(i)))
+    temp = num + i
+    if temp == target:
+        print(i)
+        break    #여기서부터 어떻게 하느냐에 따라 출력물이 매우 달랐다. 이 점을 유위하고 다시 복습하도록 하자. 
+else:
+    print("0")
```

Comment:
```suggestion
        break
        else:
            print("0")
```

오 여기 이렇게 고치면 오히려 틀리네?

궁금해서 내 환경에서 216 넣고 돌려 봤는데 결과가
```
0
0
0
0
...
198
``` 
이렇게 나오네

신기한 문법 형태라 나도 처음 보는데 원칙대로 생각하면
`if` - `else`는 쌍둥이, `if`가 틀리면 `else`에 있는 연산을 실행,
하지만 `for`에 속한 건 only `if`문. 그래서 `if`문은 계속해서 검사가 되지만 `else`문은 for loop의 마지막 한번에서만 수행됨

그래서 정답이 아닌 (분해합을 계산하며 숫자를 늘리는 과정)때에는 0을 출력하지 않기 때문에 정답

~너 이거 구글링 했냐???? 어케 이런 생각을 ㄷㄷ~